File: go/rna-transcription/rna_transcription.go
package strand

var RNA_CONVERSION = map[rune]string {'A': "U", 'T': "A", 'C': "G", 'G': "C"}

func ToRNA(dna string) string {
	rna := ""

	for _, dna_nucleotide := range dna {
		rna += RNA_CONVERSION[dna_nucleotide]
	}

	return rna
}

File: typescript/rna-transcription/rna-transcription.ts
class Transcriptor {

    private translation = new Map()

    constructor() {
        this.translation.set('A', 'U')
        this.translation.set('T', 'A')
        this.translation.set('C', 'G')
        this.translation.set('G', 'C')
    }

    toRna(dna: string) {
        let rna = ""

        for (const char of dna) {
            if (!this.translation.has(char)) {
                throw new Error('Invalid input DNA.')
            }

            rna += this.translation.get(char)
        }

        return rna
    }
}

export default Transcriptor
